lib/krikri/harvesters/couchdb_harvester.rb

```diff
@@ -35,34 +36,79 @@ def initialize(opts = {})
     # @see Analysand::StreamingViewResponse
     def record_ids(opts = {})
       view = opts[:view] || @opts[:view]
-      client.view(view, include_docs: false, stream: true).keys.lazy
+      # The set of record ids is all of the record IDs in the database minus
+      # the IDs of CouchDB design documents.
+      view_opts = {include_docs: false, stream: true}
+      client.view(view, view_opts).keys.lazy.select do |k|
+        !k.start_with?('_design')
+      end
     end
 
     ##
-    # Returns the total number of documents reported by a CouchDB view.
+    # Return the total number of documents reported by a CouchDB view.
+    #
+    # @param opts [Hash]  Analysand::Database#view options
+    #                     - view:  database view name
+    # @return [Fixnum]
+    #
     def count(opts = {})
       view = opts[:view] || @opts[:view]
-      client.view(view,
-                  limit: 0,
-                  include_docs: false,
-                  stream: true).total_rows
+      # The count that we want is the total documents in the database minus
+      # CouchDB design documents.  Asking for the design documents will give us
+      # the total count in addition to letting us determine the number of
+      # design documents.
+      v = client.view(view,
+                      include_docs: false,
+                      stream: false,
+                      startkey: '_design',
+                      endkey: '_design0')
+      total = v.total_rows
+      design_doc_count = v.keys.size
+      total - design_doc_count
     end
 
     ##
-    # Streams a response from a CouchDB view to yield documents.
+    # Makes requests to a CouchDB view to yield documents.
     #
     # The following will only send requests to the endpoint until it
     # has 1000 records:
     #
     #     records.take(1000)
     #
+    # Batches of records are requested, in order to avoid using
+    # `Analysand::StreamingViewResponse`, and the CouchDB `startkey` parameter
+    # is used for greater efficiency than `skip` in locating the next page of
+    # records.
+    #
+    # @return [Enumerator]
     # @see Analysand::Viewing
-    # @see Analysand::StreamingViewResponse
+    # @see http://docs.couchdb.org/en/latest/couchapp/views/collation.html#all-docs
     def records(opts = {})
       view = opts[:view] || @opts[:view]
-      client.view(view, include_docs: true, stream: true).docs.lazy.map do |r|
-        @record_class.build(mint_id(r['_id']), r.to_json, 'application/json')
+      limit = opts[:limit] || @opts[:limit]
+
+      en = Enumerator.new do |e|
```

Comment:
I'm thinking about ways we might be able to break this out into constituent methods.  Maybe targeting something like this as a starting place?

```ruby
def records (opts = {})
  view = opts[:view] || @opts[:view]
  limit = opts[:limit] || @opts[:limit]

  paged_rows.map do |row|
    @record_class.build(mint_id(row['doc']['_id']), row['doc'].to_json, 'application/json')
  end
end

def paged_rows(opts)
  en = Enumerator.new do |e|
    rows_retrieved = 0
    total_rows = nil
    loop do
      v = client.view(view, view_opts)
     total_rows ||= v.total_rows
     v.rows.each do |row|
        next if row['id'].start_with('_design')
        e.yield row
     end
    end
  end
end
```